snowpack/src/commands/dev.ts

```diff
@@ -427,6 +428,20 @@ export async function startServer(
     // but as a general rule all URLs contained within are managed by the package source loader. When this URL
     // prefix is hit, we load the file through the selected package source loader.
     if (reqPath.startsWith(PACKAGE_PATH_PREFIX)) {
+      // Backwards-compatable redirect for legacy package URLs: If someone has created an import URL manually
+      // (ex: /_snowpack/pkg/react.js) then we need to redirect and warn to use our new API in the future.
+      if (reqUrl.split('.').length <= 2) {
+        if (!warnedDeprecatedPackageImport.has(reqUrl)) {
+          logger.warn(`(${reqUrl}) Deprecated manual package import. Please use getUrlForPackage() to create package URLs instead.`);
+          warnedDeprecatedPackageImport.add(reqUrl);
+        }
+        const redirectUrl = await pkgSource.resolvePackageImport(
+          path.join(config.root, 'package.json'),
+          reqUrl.replace(PACKAGE_PATH_PREFIX, '').replace(/\.js/, ''),
+          config,
+        );
+        reqPath = decodeURI(url.parse(redirectUrl).pathname!)
+      }
```

Comment:
Adding this "redirect" to the `loadUrl` handler, as that's where JS API users will likely encounter this issue. It's also called internally by `serverRuntime.importModule()`.

Could be extracted to a shared function but it's pretty small.

Reply:
Yea, fair enough. This does make me a bit nervous (returning a response body for a different URL than was asked for, since unique URLs mean so much in this world) which is why I implemented it in the handleRequest handler instead. But your point about most users using this API and not the other is important. Even if this is a bit buggy in some edge cases, its better to warn all users and the warning will make the cause a bit more clear.

Reply:
Very true. Without "redirecting" here, `loadUrl` users were just getting a `NotFound` error with no message. Would throwing a `NotFound` error with a message explaining the breaking change be better?